src/frontend/src/views/Account/index.tsx

```diff
@@ -0,0 +1,125 @@
+import { Button, Icon, InputText, Link } from "czifui";
+import { useRouter } from "next/router";
+import React, { ChangeEvent, useEffect, useState } from "react";
+import { H1, H2, P } from "src/common/styles/basicStyle";
+import AbandonChangesModal from "./components/AbandonChangesModal";
+import {
+  GrayIconWrapper,
+  StyledDivider,
+  StyledH3,
+  StyledHeaderRow,
+  StyledRow,
+  StyledSection,
+  SubText,
+  WhiteIconWrapper,
+} from "./style";
+
+export default function Account(): JSX.Element {
+  const router = useRouter();
+  // TODO:194969 - get user's current gisaid id for the default value
+  const [gisaidId, setGisaidId] = useState("");
+  const [hasUnsavedChanges, setUnsavedChanges] = useState<boolean>(false);
+
+  const [isModalOpen, setModalOpen] = useState<boolean>(false);
+
+  // TODO: 194969 - implement save once API is ready
+  const handleSave = () => {
+    // eslint-disable-next-line
+    console.log(gisaidId);
+    setUnsavedChanges(false);
+  };
+
+  const handleNewIdInput = (e: ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setGisaidId(value);
+    setUnsavedChanges(true);
+  };
+
+  const openModal = () => {
+    setModalOpen(true);
+  };
+
+  const closeModal = () => {
+    setModalOpen(false);
+  };
+
+  // prompt the user if they try and leave with unsaved changes
+  useEffect(() => {
+    const handleWindowClose = (e: BeforeUnloadEvent) => {
+      if (!hasUnsavedChanges) return;
+      e.preventDefault();
+      openModal();
+      return false;
+    };
+    const handleBrowseAway = () => {
+      if (!hasUnsavedChanges) return;
+      openModal();
+      router.events.emit("routeChangeError");
+      throw "routeChange aborted.";
+    };
+    window.addEventListener("beforeunload", handleWindowClose);
+    router.events.on("routeChangeStart", handleBrowseAway);
+    return () => {
```

Comment:
👏  love this

Reply:
Unfortunately, I haven't been able to figure out how to get the modal to do the action once the user decides to leave.  For route changes, I could pass in the route and have the modal redirect them.  However, for closing the page, this doesn't seem to work. `window.close()` apparently only works if we opened the window in the first place. Do you know if it's possible to close the page for them?

Reply:
Do you mean in handleWindowClose? If you remove the e.preventDefault, I think the default event handler also runs after your custom code _if_ no other handlers catch it or cancel it. Does removing that line generate the desired effect?

Reply:
It got weird with mounting a new component.  I ended up not worrying about it per [this comment](https://czi-sci.slack.com/archives/C01HYJYA50W/p1659719347193429?thread_ts=1659715202.644959&cid=C01HYJYA50W) from Janeece.